apps/analytics/views.py

```diff
@@ -107,15 +108,21 @@ def get_submission_count(request, challenge_pk, duration):
 @api_view(['GET', ])
 @permission_classes((permissions.IsAuthenticated, HasVerifiedEmail, IsChallengeCreator))
 @authentication_classes((ExpiringTokenAuthentication,))
-def get_challenge_phase_submission_analysis(request, challenge_pk, challenge_phase_pk):
+def get_challenge_phase_submission_analysis(request, challenge_pk, challenge_phase_id):
     """
     API to fetch
     1. The submissions count for challenge phase.
     2. The participated team count for challenge phase.
     """
     challenge = get_challenge_model(challenge_pk)
 
-    challenge_phase = get_challenge_phase_model(challenge_phase_pk)
+    try:
+        challenge_phase = ChallengePhase.objects.get(
+            challenge=challenge, phase_id=challenge_phase_id)
+    except ChallengePhase.DoesNotExist:
+        response_data = {
+            'error': 'Challenge Phase {} does not exist'.format(challenge_phase_id)}
+        return Response(response_data, status=status.HTTP_404_NOT_FOUND)
 
```

Comment:
@isht3 Please reduce this redundant code everywhere by compiling it into a function. For reference please take a look at https://github.com/Cloud-CV/EvalAI/blob/master/apps/base/utils.py#L43

Reply:
I tried and it didn't work. Since the Response is returned when the ChallengePhase does not exist, it throws an error. And if we only take the get statement out we cannot do the error handling.